substra/sdk/client.py

```diff
@@ -49,26 +56,158 @@ def wrapper(*args, **kwargs):
     return wrapper
 
 
+def _upper_slug(s: str) -> str:
+    return slugify(s, separator="_").upper()
+
+
+def _get_env_var(attribute_name: str, configuration_name: str) -> str:
+    """
+    Returns the environment variable associated with a given attribute for a given Client,
+    identified by its configuration_name
+    """
+    return f"SUBSTRA_{_upper_slug(configuration_name)}_{_upper_slug(attribute_name)}"
+
+
+@dataclasses.dataclass
+class SettingValue:
+    value: Any
+    origin: Literal["code", "environment_variable", "config_file", "default"]
+
+
+def _get_config_value(
+    *,
+    attribute_name: str,
+    code_values: Dict[str, Any],
+    configuration_name: Optional[str],
+    client_config_dict: Dict[str, str],
+    mapping: Dict[str, callable],
+) -> SettingValue:
+    """
+    For a given attribute, return the value to be used in the Client configuration.
+    The order of precedence is: values defined by the user in the code, environment variables, values read from the
+    configuration file. If the attribute is not set, the value returned is None (and the origin is set to "default").
+
+    Args:
+        attribute_name (str): Name of the attribute for which we want to get the value to use
+        code_values (dict): Dictionary containing the values read from the code (or None)
+        configuration_name (str or None): Name of the configuration to use; it is used to determine which env var to use
+            and which sub-dictionary of the configuration file to consider; if None, only the values defined in the code
+            are considered.
+        client_config_dict (dict): Content of the configuration file
+        mapping (dict): Mapping associating a callable to each attribute; usually used for type conversion, but could be
+            used for more advanced processing
+
+    Returns:
+        SettingValue: Object containing the value and its origin (code, environment_variable, config_file or default)
+    """
+
+    # Some values are boolean that might be set to False, so we check explicitly "is not None"
+    # Note that explicitly passing "None" in the code is equivalent to omitting it.
+    # (code_values will actually always contain all the keys, with None values for keys that were omitted)
+    if code_values.get(attribute_name) is not None:
+        return SettingValue(value=code_values[attribute_name], origin="code")
+
+    # if client_name is not given, we cannot look through env var and config files
+    if not configuration_name:
+        return SettingValue(value=None, origin="default")
```

Comment:
If we raise the error, we can remove this part

Reply:
I don't think we can because it's still acceptable to configure a `Client` entirely from code (and so without any configuration_name). 
I think it's a bit clearer to quit at that point.